### PR TITLE
Control Header Object

### LINKED ISSUE #1: add a header class for the control
- [ ] the header should be a blog of text appened to the top of the to_ruby file and a sstandard key value in the to_hash
- [ ] MVP is just plain text - no fancy input or string interpolation

```ruby 
# START HEADER
## ruby bock of text that just appends to the file, may or may not support string intropolation 
## or evauation of inspec inputs
# END HEADER
control "V-71849" do
  title "The Red Hat Enterprise Linux operating system must be configured so
  that the file permissions, ownership, and group membership of system files and
  commands match the vendor values."
  
  desc  "Discretionary access control is weakened if a user or group has access
  permissions to system files and directories greater than the default.
  ...
end
```
### Result
```ruby 
# my company header text 
# -*- encoding : utf-8 -*-
control "V-71849" do
  title "The Red Hat Enterprise Linux operating system must be configured so
  that the file permissions, ownership, and group membership of system files and
  commands match the vendor values."
  
  desc  "Discretionary access control is weakened if a user or group has access
  permissions to system files and directories greater than the default.
  ...
end
```

### PR DESCRIPTION
**Pull request by aaronlippold**
- added basic header object - implementation is a basic string block
- added unit tests

Fixes #1 

Signed-off-by: Aaron Lippold <lippold@gmail.com>

### CODE REVIEW

File: lib/inspec/objects/control.rb [hunk 1]
```diff
@@ -18,21 +25,28 @@ def add_tag(t)
...
-    def to_ruby # rubocop:disable Metrics/AbcSize
-      res = ["control #{id.inspect} do"]
+    def to_ruby
+      if header.empty? || header.nil? || header == []
+        res = ["control #{id.inspect} do"]
+      else
+        res = []
+        res.push header
+        res.push "control #{id.inspect} do"
+      end
```
Comment (timwongj) [inline lib/inspec/objects/control.rb:45]:
```suggestion
      res = []
      res.push header if header.present?
      res.push "control #{id.inspect} do"
```

File: .gitignore [hunk 2]
```diff
@@ -48,3 +48,4 @@ Gemfile.lock
 
 # unless supporting rvm < 1.11.0 or doing something fancy, ignore this:
 .rvmrc
+.vscode/*
```
Comment (timwongj) [inline .gitignore:51]: why is this removed?

**Review by timwongj — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (aaronlippold) [reply .gitignore:51]: it's not, I told it to ignore that dir if it exists
